master/internal/trials/api_trials.go

```diff
@@ -55,16 +56,27 @@ func checkTrialFiltersEmpty(f *apiv1.TrialFilters) error {
 func (a *TrialsAPIServer) QueryTrials(ctx context.Context,
 	req *apiv1.QueryTrialsRequest,
 ) (*apiv1.QueryTrialsResponse, error) {
-	err := checkTrialFiltersEmpty(req.Filters)
+	curUser, _, err := grpcutil.GetUser(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("error querying for trials %w", err)
+	}
+	err = checkTrialFiltersEmpty(req.Filters)
 	if err != nil {
-		return nil, fmt.Errorf("error querying tags for trials %w", err)
+		return nil, fmt.Errorf("error querying for trials %w", err)
```

Comment:
This looks like an error that should happen when a user sends a bad request. I recommend using `status.Error` [like this](https://github.com/determined-ai/determined/blob/master/master/internal/usergroup/api_groups.go#L26) rather than sending back a 500 for every error.

It's worth going through the errors returned from `TrialsAPIServer` and making sure they're as useful as they can be.

Reply:
it turns out this was wrapping a`status.Error` so I just unwrapped it.

most of errors returned by `TrialsAPIServer` are just propagated from bun, those I left as 500s, since I don't know why they would occur anyways.